ckanext/ontario_theme/resource_upload.py

```diff
@@ -169,9 +174,16 @@ def _check_zip_mimetype(self):
                       
         elif self.clear:
             resource['url_type'] = ''
-
-        if url and not (resource.get('url_type') == 'upload') and not resource.get('format'):
+        
+        # Format should be WEB for all remote resources regardless of
+        # the actual type
+        if url and not (resource.get('url_type') == 'upload'):
             resource['format'] = 'WEB'
 
-        if resource.get('format'):
-            resource['format'] = resource.get('format').upper()
+        # For uploaded resources, use the specified format unless it does
+        # not agree with the format derived from the guessed mime type
+        if resource.get('url_type') == 'upload' and resource.get('format'):
+            if this_format and resource.get('format').lower() != this_format:
+                resource['format'] = this_format
```

Comment:
`this_format` is always upper case because it derives from the first column of this file (https://github.com/ongov/ckanext-ontario_theme/blob/main/ckanext/ontario_theme/accepted_resource_formats.json), which is always upper case.